aten/src/ATen/cuda/CUDABlas.cpp

```diff
@@ -2124,7 +2124,7 @@ void vdot<c10::complex<double>>(CUDABLAS_DOT_ARGTYPES(c10::complex<double>)) {
                                    incx, reinterpret_cast<const cuDoubleComplex*>(y), incy,
                                    reinterpret_cast<cuDoubleComplex*>(result)));
 }
-
+#if !defined(USE_ROCM) && !defined(_MSC_VER)
```

Comment:
Is this logic correct?  You want to disable this code for USE_ROCM && _MSC_VER because HIP on Windows doesn't support sparse.  But all other prior cases should remain.  Meaning, ROCm on linux, CUDA on linux and windows.
```suggestion
#if !(defined(USE_ROCM) && defined(_MSC_VER))
```
Or
```
#if !defined(USE_ROCM) || !defined(_MSC_VER)
```

Reply:
Thanks for the review and suggestion. Yes, we want to disable for just ROCm on Windows for now.